### PR TITLE
Add GitHub workflows for CI and Release

### PR DESCRIPTION
**Pull request by nickelc**
The CI workflow combines the Travis & Appveyor setup except for the release builds. 
These are moved into the Release workflow.

related: #543

### CODE REVIEW

**Comment by ozkriff** [comment]
Wow, thanks for the PR! Could you please rebase on top of the master branch?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+jobs:
+  lints:
+    name: Rustfmt and clippy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install packages
```
Comment (ozkriff) [inline .github/workflows/ci.yml]: Hmm, why can't this be folded into a `packages: libasound2-dev libudev-dev` line?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+          echo "Assets: $ASSETS"
+          if [ "$SRC" != "$ASSETS" ]; then
+            false
+          fi
+
+  build:
+    name: ${{ matrix.build }}
+    runs-on: ${{ matrix.os }}
+
+    # The build matrix does not yet support 'allow failures' at job level.
```
Comment (ozkriff) [inline .github/workflows/ci.yml]: Hmm.. Nightly failures still mark the whole commit as "not successful".

image

Is it possible to do something about it?

Comment (ozkriff) [reply .github/workflows/ci.yml]: https://github.community/t5/GitHub-Actions/continue-on-error-allow-failure-UI-indication/td-p/37033 :(

Comment (nickelc) [reply .github/workflows/ci.yml]: Yes, you can disable `fail-fast` in the build matrix.

Comment (nickelc) [reply .github/workflows/ci.yml]: Yes, you can configure the required status checks under `Settings > Branches`
image

Comment (ozkriff) [reply .github/workflows/ci.yml]: Yeah, I've already configured this and I'm able to merge the PR. It just troubles me that the commit is marked as ❌ - it feels somewhat wrong.

Comment (nickelc) [reply .github/workflows/ci.yml]: Ok, i've tested it https://github.com/nickelc/zemeroth/pull/1 and `continue-on-error` for nightly is the only way to keep it green at the moment.

Comment (ozkriff) [reply .github/workflows/ci.yml]: Could you please add it to this PR?

**Comment by ozkriff** [comment]
Also, could you please add a status badge to the readme?

https://img.shields.io/github/workflow/status/ozkriff/zemeroth/CI/master?label=GithubActions should work after the PR is merged (i guess)

**Comment by nickelc** [comment]
> Also, could you please add a GA status badge to the readme?

Do you want the shields.io badge or the badge from [github](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/configuring-a-workflow#adding-a-workflow-status-badge-to-your-repository)?

**Comment by ozkriff** [comment]
Either one is fine with me

**Comment by ozkriff** [comment]
Awesome, thanks for the PR! :)

**Pull request merged**